docs/src/content/docs/basic-concepts.mdx

```diff
@@ -10,11 +10,16 @@ GenSX is a simple typescript framework for building complex LLM applications. It
 Components are the building blocks of GenSX applications; they're pure TypeScript functions that:
 
 - Accept props as input
-- Produce an output
+- Produce an output that can be used by other components or returned as the result of the workflow
 - Don't depend on global state
 - Are strongly typed using TypeScript
 
-Components can return data as well as other components. Their outputs can be used by other components or returned as the result of the workflow.
+When you define a component, you need to include four things:
+
+1. The input type, or props, for the component
+2. The output type for the component
+3. The name of the component that's used to give the component a unique identifier
```

Comment:
@EvanBoyle - any opinions on how we describe the name? i struggled to explain it since it's not technically used anywhere yet iiuc

Reply:
The name of the component that's used for telemetry, tracing, visualization, and caching.

Reply:
ahh much better